methods.py

```diff
@@ -24,7 +24,10 @@ def add_source_files(self, sources, files):
             dir_path = self.Dir(".").abspath
             files = sorted(glob.glob(dir_path + "/" + files))
             if skip_gen_cpp:
-                files = [f for f in files if not f.endswith(".gen.cpp")]
+                if allow_gen:
+                    files = [f for f in files]
```

Comment:
That's basically `files = files` :)

Instead if `allow_gen`, then you shouldn't even do the `skip_gen_cpp` logic.